src/CommunityToolkit.Maui/Views/Popup/PopupLifecycleController.cs

```diff
@@ -1,38 +1,28 @@
-using CommunityToolkit.Maui.Core;
 using CommunityToolkit.Maui.Views;
 
 namespace CommunityToolkit.Maui;
 
 /// <summary>
 /// List-based implementation that manages the presentation of popups which will return the last item as the current popup.
 /// </summary>
-public class PopupLifecycleController : IPopupLifecycleController
+class PopupLifecycleController
 {
-	readonly List<WeakReference<Popup>> currentPopups = [];
+	readonly List<WeakReference<PopupContainer>> currentPopups = [];
```

Comment:
There're scenarios in .net maui that `Window` and `Application` will recreated (if user doesn't cache them) which can cause this `List` to increase with invalid `WeakReference`, would be good to have a logic that will lookup into the list and remove invalid `WeakReferences`